sdk/template/.template.config/template.json

```diff
@@ -328,43 +332,44 @@
             "value": "."
           }
         ],
-        "separator": ""
+        "separator": "",
+        "removeEmptyValues": true
       },
       "fileRename": "Azure.Template."
     },
     "YamlServiceDirectory": {
       "type": "generated",
       "generator": "join",
       "parameters": {
-          "symbols": [
-              {
-                  "type": "const",
-                  "value": "ServiceDirectory: "
-              },
-              {
-                  "type": "ref",
-                  "value": "ServiceNameLowercase"
-              }
-          ],
-          "separator": ""
+        "symbols": [
+          {
+            "type": "const",
+            "value": "ServiceDirectory: "
+          },
+          {
+            "type": "ref",
+            "value": "ServiceNameLowercase"
+          }
+        ],
+        "separator": ""
       },
       "replaces": "ServiceDirectory: template"
     },
     "AzureResourceProviderNamespace": {
       "type": "generated",
       "generator": "join",
       "parameters": {
-          "symbols": [
-              {
-                  "type": "const",
-                  "value": "Microsoft"
-              },
-              {
-                  "type": "ref",
-                  "value": "ServiceNameCapitalCase"
-              }
-          ],
-          "separator": "."
+        "symbols": [
+          {
+            "type": "const",
+            "value": "Microsoft"
+          },
+          {
+            "type": "ref",
+            "value": "ServiceNameCapitalCase"
+          }
+        ],
+        "separator": "."
```

Comment:
Shall we add `"removeEmptyValues": true` here also?

Reply:
I suppose it couldn't hurt, but may not be needed if you found everything else worked. You should test again with my update just to make sure.